ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/MeteredLedgerDao.scala

```diff
@@ -0,0 +1,312 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
```

Comment:
Do we need to duplicate this class? I am not convinced.

Reply:
Good point, it is identical. Although there are multiple things here which is questionable regarding duplication, but I deliberately did not wanted to move in that territory much at the moment, so we have potentially easier hand later on.
TLDR: I would limit "reuse" or most importantly dependency between forked and original code.
Are you okay to draw the line here, and keep the metered dao duplicated?

Reply:
In case of this particular class it is more difficult to keep a duplicate than reuse. It is implemented using a decorator pattern so it will ALWAYS follow the interface defined in LedgerDao and no more or less.